src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -793,6 +868,23 @@ protected ExecutionResult handleNonNullException(ExecutionContext executionConte
         return executionResult;
     }
 
+    protected void handleInnerNonNullException(ExecutionContext executionContext, CompletableFuture<Object> result, Throwable e) {
+        Throwable underlyingException = e;
+        if (e instanceof CompletionException) {
+            underlyingException = e.getCause();
+        }
+        if (underlyingException instanceof NonNullableFieldWasNullException) {
+            assertNonNullFieldPrecondition((NonNullableFieldWasNullException) underlyingException, result);
+            if (!result.isDone()) {
+                result.complete(null);
+            }
+        } else if (underlyingException instanceof AbortExecutionException) {
+            result.complete(null);
+        } else {
+            result.completeExceptionally(e);
+        }
+    }
```

Comment:
Signature change for a method that is otherwise a copy.

NOTE:   I think I found a problem in the existing code - if we get an `NonNullableFieldWasNullException` inside an inner object or a list - I think it gets lost or it was always redundant - either way we should investigate this.

That is have a inner list produce a null value for a non null field and see what happens with todays code